### PR TITLE
Student Scores Concept Coach update table layout and cell data

### PR DESCRIPTION
**Pull request by darrenmason**
image

http://3lwsnp.axshare.com/#p=detailed_scores

- [x] new header structure/labels
- [x] progress icons / split cells
- [x] display as
- [x] based on (only renders if data_headings.average_possible is present)
- [x] sort by score or completed
- [x] sortable header hover
- [x] new icons
- [x] total attempted/possible data
- [x] refactored selenium test

refs openstax/tutor-server#978

### CODE REVIEW

File: src/components/scores/pie-progress.cjsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+React  = require 'react'
+
+PieProgress = React.createClass
```
Comment (philschatz) [inline src/components/scores/pie-progress.cjsx]: Cool pie chart!

File: src/components/scores/pie-progress.cjsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      </svg>
+    finished =
+      <svg className='finished'>
+        <path
+        d="M12 0C5.372 0 0 5.373 0 12c0 6.627 5.372 12 12 12c6.628 0 12-5.373 12-12C24 5.373 18.628 0 12 0z
+          M10.557 19.455l-7.042-7.042l2.828-2.828l4.243 4.242l7.07-7.071l2.829 2.829L10.557 19.455z" />
+      </svg>
+    notStarted = <i className="fa fa-minus"/>
+    
+    if value >= 100
```
Comment (philschatz) [inline src/components/scores/pie-progress.cjsx]: Would it make sense to add console logging or something if the numbers are `> 100` or `< 0`?

**Comment by philschatz** [comment]
When I sort by scores I get the following:

image


It seems unstarted is treated the same as started but 0 correct answers; should that be the case?

Also, isn't there supposed to be a 2nd toggle for attempted vs assigned?

**Comment by darrenmason** [comment]
The 2nd toggle for attempted vs assigned is finished but not being displayed due to missing data.  Should I show it with blank cells instead?

File: src/components/scores/pie-progress.cjsx (hunk 2 above)
Comment (darrenmason) [inline src/components/scores/pie-progress.cjsx]: yes, I can add error log

**Comment by darrenmason** [comment]
regarding sort, I may need to sort by status 'not_started' as well

File: src/components/scores/pie-progress.cjsx (hunk 2 above)
Comment (philschatz) [inline src/components/scores/pie-progress.cjsx]: Thoughts on error logging? I was thinking we _could_ log if in development mode and then silently recover in production. Thoughts?

/cc @nathanstitt @pandafulmanda

Comment (pandafulmanda) [inline src/components/scores/pie-progress.cjsx]: yes i think all our logs should do this, and perhaps we should have a wrapper log function or lib for this.  I'd searched for one once or twice

Comment (nathanstitt) [inline src/components/scores/pie-progress.cjsx]: Also, the BE now has a logging endpoint https://github.com/openstax/tutor-server/pull/909 It'd be nice to log to that, but it'd be more work to setup

File: src/components/scores/pie-progress.cjsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    else
+      75
+
+  render: ->
+    {size, value, roundToQuarters} = @props
+    radius = @radius(size)
+    fullCircle = <circle r="#{radius}" cx="#{radius}" cy="#{radius}" className='slice'></circle>
+    circle = if roundToQuarters? then @buildCircle(@roundToQuarters(value)) else @buildCircle(value)
+    path = <path d="#{circle}" className='slice' />
+    pieCircle =
```
Comment (philschatz) [inline src/components/scores/pie-progress.cjsx]: Is this case shown in a screenshot?

**Comment by philschatz** [comment]
Are the current averages from BE really based on the attempted count? I thought it would be based on the total possible.

/cc @nathanstitt

**Comment by nathanstitt** [comment]
It's based on attempted count until the task is past due.  After that point it's based on the total possible.  I believe that's because the student could always come back later and finish the rest of the steps.

[BE algorithm](https://github.com/openstax/tutor-server/blob/78507e4a54a2c5e2fa131b1b64a4d1c76eba8fdf/app/subsystems/tasks/performance_report_methods.rb#L6-L21)

**Comment by darrenmason** [comment]
updated averages.  depends on : https://github.com/openstax/tutor-server/pull/978

File: src/components/scores/table-hs.cjsx [hunk 4]
```diff
@@ -90,10 +90,10 @@ module.exports = React.createClass
           Review
         </Router.Link>
 
-    if heading.average
+    if heading.attempted_average
```
Comment (Dantemss) [inline src/components/scores/table-hs.cjsx]: Not sure if this matters, but we currently have the equivalent of total_average here (in production).

**Comment by philschatz** [comment]
:sunglasses: getting the selenium tests in!

File: test-integration/npm-debug.log [hunk 5]
```diff
@@ -0,0 +1,18 @@
+0 info it worked if it ends with ok
```
Comment (philschatz) [inline test-integration/npm-debug.log]: Did you mean to include this file?

File: test-integration/scores.coffee [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  afterEach ->
+    @user.goHome()
+
+
+  @it 'sorts by name or data', ->
+    @scores.el.nameHeaderSort.get().click()
+    @scores.el.dataHeaderSort.get().click()
+
+  @it 'changes periods', ->
```
Comment (philschatz) [inline test-integration/scores.coffee]: I wonder if some selenium running time could be saved by combining these into 1 test; thoughts?

File: test-integration/helpers/scores.coffee [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    super(testContext, testElementLocator, COMMON_ELEMENTS)
+    @setCommonHelper('periodReviewTab', new PeriodReviewTab(@test))
+
+  # this could be moved to cc-dashboard helper at some point
+  goCCScores: =>
+    @test.utils.windowPosition.scrollTop()
+    @el.ccScoresLink.get().click()
+
+  doneGenerating: =>
```
Comment (philschatz) [inline test-integration/helpers/scores.coffee]: Could this be renamed to `isDoneGenerating` to clarify that it is a test that yields a boolean?

File: test-integration/scores.coffee (hunk 6 above)
Comment (darrenmason) [inline test-integration/scores.coffee]: yes, was planning on that, need to detect CC after login, will do

File: test-integration/helpers/scores.coffee (hunk 7 above)
Comment (darrenmason) [inline test-integration/helpers/scores.coffee]: makes sense.

File: test-integration/npm-debug.log (hunk 5 above)
Comment (darrenmason) [inline test-integration/npm-debug.log]: no, oops :)

**Pull request merged**